### PR TITLE
Ether transfer

### PR DESCRIPTION
**Pull request by ptsayli**

### CODE REVIEW

File: src/common/ContractsBase.ts [hunk 1]
```diff
@@ -34,23 +35,23 @@ export default class ContractsBase {
     return new web3.eth.Contract(ChildERC721Artifact.abi, token)
   }
 
+  public getChildMaticContract(childMaticAddress: address) {
+    return new this.web3Client.web3.eth.Contract(ChildMaticArtifact.abi, childMaticAddress)
```
Comment (atvanguard) [inline src/common/ContractsBase.ts]: `childMaticAddress` is fixed. Can we make that a constant and use it here?

File: examples/transfer-ether.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+const bn = require('bn.js')
+
+const utils = require('./utils')
+
+async function execute() {
+  const { matic, network } = await utils.getMaticClient('testnet', 'v3')
+  matic.setWallet(utils.getPrivateKey())
+
+  const amount = new bn(10).pow(new bn(17)) // 0.1 token
+  return matic.transferEthers(process.env.FROM, amount, { from: process.env.FROM })
```
Comment (atvanguard) [inline examples/transfer-ether.js]: Add a comment saying, `// performing a self-transfer` so it's clearer what's happening here.

File: src/index.ts [hunk 3]
```diff
@@ -115,6 +115,32 @@ export default class Matic extends ContractsBase {
...
+      throw new Error('Missing Parameters')
+    }
+    const from = options.from
+    const value = this.encode(amount)
+    if (!options.parent) {
+      const maticWeth = await this.registry.registry.methods.getWethTokenAddress().call()
+      return this.transferERC20Tokens(maticWeth, toAddress, value, options)
+    }
+    const web3Object = this.web3Client.getParentWeb3()
+    const gasLimit = await web3Object.eth.estimateGas({
```
[the remark this answers is not in the document]
Comment (ptsayli) [reply src/index.ts]: if gas value is already set by user that will be handled in `_fillOptions` function

File: src/common/ContractsBase.ts (hunk 1 above)
Comment (ptsayli) [reply src/common/ContractsBase.ts]: actually, I required token address if the user wants raw data, so passing it as param
So basically It is declared const from where this function is called

File: src/index.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (atvanguard) [reply src/index.ts]: Incorrect. Both `options.gas = gasLimit` and `Object.assign(options, { gas, value, to })` will override the user input for gas. Moreover, we should avoid the `estimateGas` called if user has already sent in the gas amount.

[the remark this answers is not in the document]
Comment (ptsayli) [reply src/index.ts]: oops
My bad

**Pull request merged**